From 4e6ed23145ce6deb4710b781b94cca31e9fcff49 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 21 Apr 2025 12:19:06 +0800
Subject: [PATCH] helpers.h: fix check for st_birthtimespec on macOS

This should be based on system version, not bitness of the arch.
Notice, this condition is true for all released macOS versions,
but not for some pre-release builds of 10.6, where st_birthtimespec
is missing. There is no easy way to check for those; if you build
on 10a96 or 10a190, please modify the macro accordingly.
---
 src/helpers.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/helpers.h b/src/helpers.h
index bd47fd4b..d557b423 100644
--- src/helpers.h
+++ src/helpers.h
@@ -400,9 +400,13 @@ if (S_ISNWK(mode)) return 'n'; // HP/UX: network special file
  * actually store creation times: the value of __st_birthtim is always zero.
 #elif defined(__OpenBSD__)
 # define ST_BTIME __st_birthtim */
-# elif defined(__NetBSD__) || (defined(__APPLE__) && !defined(__ppc__) \
-&& !defined(__i386__))
+# elif defined(__NetBSD__)
 #  define ST_BTIME st_birthtimespec
+# elif defined(__APPLE__)
+#  include <AvailabilityMacros.h>
+#  if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#   define ST_BTIME st_birthtimespec
+#  endif
 # elif defined(__FreeBSD__) || defined(__CYGWIN__)
 #  define ST_BTIME st_birthtim
 # elif defined(__HAIKU__)
